src/models/multiBarChart.js

```diff
@@ -179,11 +179,6 @@ nv.models.multiBarChart = function() {
             if (showLegend) {
                 legend.width(availableWidth - controlWidth());
 
-                if (multibar.barColor())
-                    data.forEach(function(series,i) {
-                        series.color = d3.rgb('#ccc').darker(i * 1.5).toString();
-                    });
```

Comment:
What I describe below happens in the HorizontalBarChart, which is the one I am using, but I assume the behaviour is the same as the code is almost identical.

The conditional evaluates to the function, even when chart.barColor(false) is called when creating a chart.

printing out the result of multibar.barColor() gives this (just in case it's relevant somehow)
```
function (d, i) {
    return d.color || colors[i % colors.length]
}
```
Because the conditional evaluates to a truthy value, the forEach is executed, which overwrites the values of all the series in the chart. This is not the desired behaviour as (in my case) series.color is already populated with the appropriate color.

It means that regardless of calling chart.barColor(false) and providing a custom colouring function in chart.color(...), the resulting chart will always be in chades of gray, see below:
![screen shot 2015-01-09 at 19 47 39](https://cloud.githubusercontent.com/assets/7631584/5686281/8203820e-9839-11e4-9882-47821337db56.png)

When lines 182-185 are removed, barColor colouring is not enabled without explicitly calling chart.barColor() (somehow, I did not look into why). Not explicitly calling either chart.barColor() or chart.color() makes the chart use default colouring as below:
![screen shot 2015-01-09 at 19 51 28](https://cloud.githubusercontent.com/assets/7631584/5686361/09e7098e-983a-11e4-8932-8c7b3b6ca0e0.png)
Explicitly setting the colouring function with chart.color() produces the desired result (i.e. by series colouring):
![screen shot 2015-01-09 at 19 48 40](https://cloud.githubusercontent.com/assets/7631584/5686368/259d90a8-983a-11e4-952a-1fee17c45925.png)
Explicitly setting barColor with chart.barColor() also produces the desired result (i.e. the barColor style colouring):
![screen shot 2015-01-09 at 20 04 02](https://cloud.githubusercontent.com/assets/7631584/5686448/c2c9ddc8-983a-11e4-830a-5fa54105eb6c.png)